### PR TITLE
[search] Using SmallSet for Locales.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

File: indexer/categories_holder.cpp [hunk 1]
```diff
@@ -143,6 +143,7 @@ void ProcessCategory(string const & line, vector<string> & groups, vector<uint32
 // static
 int8_t const CategoriesHolder::kEnglishCode = 1;
 int8_t const CategoriesHolder::kUnsupportedLocaleCode = -1;
+uint8_t constexpr CategoriesHolder::kMaxSupportedLocaleIndex;
```
Comment (ygorshenin) [inline indexer/categories_holder.cpp]: Please, be consistent - use const or constexpr, but don't mix them.

File: indexer/categories_holder.hpp [hunk 2]
```diff
@@ -65,6 +65,7 @@ class CategoriesHolder
 public:
   static int8_t const kEnglishCode;
   static int8_t const kUnsupportedLocaleCode;
+  static uint8_t constexpr kMaxSupportedLocaleIndex = 30;
```
Comment (ygorshenin) [inline indexer/categories_holder.hpp]: You can initialize in header other constants, right?

File: indexer/categories_holder.cpp (hunk 1 above)
Comment (mpimenov) [reply indexer/categories_holder.cpp]: done

File: indexer/categories_holder.hpp (hunk 2 above)
Comment (mpimenov) [reply indexer/categories_holder.hpp]: done

**Comment by mpimenov** [comment]
@ygorshenin PTAL

**Comment by mpimenov** [comment]
@tatiana-kondakova PTAL

File: search/processor.cpp [hunk 3]
```diff
@@ -398,11 +398,11 @@ Locales Processor::GetCategoryLocales() const
...
   // Prepare array of processing locales. English locale is always present for category matching.
   if (m_currentLocaleCode != -1)
-    result.push_back(m_currentLocaleCode);
+    result.Insert(static_cast<uint64_t>(m_currentLocaleCode));
   if (m_inputLocaleCode != -1 && m_inputLocaleCode != m_currentLocaleCode)
-    result.push_back(m_inputLocaleCode);
+    result.Insert(static_cast<uint64_t>(m_inputLocaleCode));
   if (enLocaleCode != m_currentLocaleCode && enLocaleCode != m_inputLocaleCode)
-    result.push_back(enLocaleCode);
+    result.Insert(static_cast<uint64_t>(enLocaleCode));
```
Comment (tatiana-kondakova) [inline search/processor.cpp]: теперь можно не проверять `m_inputLocaleCode != m_currentLocaleCode`, `enLocaleCode != m_currentLocaleCode`, `enLocaleCode != m_inputLocaleCode`

Comment (mpimenov) [reply search/processor.cpp]: И правда.
done

Comment (tatiana-kondakova) [reply search/processor.cpp]: пока не вижу закомиченных изменений, после изменений LGTM

Comment (mpimenov) [reply search/processor.cpp]: done

**Pull request merged**